doc/spec_TransferElement.dox

```diff
@@ -239,7 +243,8 @@ This documnent is currently still **INCOMPLETE**!
 - 8.5.2 As the asynchonous mechanism and a synchronous read are two idependent channels there are potential race condition, depending on the exact protocol.
   The backend has to avoid this if possible. If it cannot be avoided, the implementation must make sure that the last value in the queue is the newest value, and this is not dopped or missed, even if the values before are not in order or send twice.
 
-- 9. It does not matter if the exception occured in an asynchronous or synchronous read, or in a write operation.
+- \anchor transferElement_comment_B_9a \ref transferElement_B_9 "9." The ChimeraTK::runtime_error is the only exception that conceptually is recoverable by calling DeviceBackend::open(). ChimeraTK::logic_error is a programming or configuration mistake. The boost::numeric::bad_numeric_cast conceptually also is a logical error (wrong data type) which only shows at run time when the data is overflowing. The boost::thread_interrupted is used when destroying the DeviceBackend object, so there is nothing to recover afterwards.
```

Comment:
> The boost::thread_interrupted is used when destroying the DeviceBackend object, so there is nothing to recover afterwards.

I think this is not the right argument. The idea is to keep the backend and accessor functional even if thead_interrupted is thrown - otherwise we could greatly simplify the specs here (just throw it completely unhandled). One could use the thread_interrupted exception also in different contexts. It is simply the equivalent of boost::thread::interrupt(). There is nothing to recover, because nothing is broken. We merely interrupt the blocking function to continue with something else.

Reply:
Can boost::thread_interrupted only be thrown as a result of calling TE::interrupt()?

Reply:
yes

Reply:
It was written at some point explicitly that it may only be thrown when interrupt() is called. This was removed when we moved it into the base class (was in section C before). Now we have 8.6 stating it is done like that, and C.1 does not list it as an allowed exception.

Reply:
OK. I re-phrased the last sentense:
```
The boost::thread_interrupted is thrown when TransferElement::interrupt() is called. It means that a blocking read() has been interrupted on request, not that the device is malfunctioning and would need recovery.
```

Reply:
I just pushed the changes